File: migratory_birds.py
birds = [1, 2, 3, 4, 5, 4, 4, 2, 1, 3, 4]

def migratoryBirds(arr):
  count_birds = dict()
  for type in arr:
    if type not in count_birds:
      count_birds[type] = 1
    else:
      count_birds[type] += 1
    
  return max(count_birds, key=count_birds.get)

print(migratoryBirds(birds))

File: migratoryBirds.js
const birds = [1, 2, 3, 4, 5, 4, 4, 2, 1, 3, 4]

function migratoryBirds(arr) {
  let countBirds = {};

  arr.forEach((count) => {
    countBirds[count] = (countBirds[count] || 0) + 1
  });

  const maxBirds = Math.max(...Object.values(countBirds));
  const keyBirds = Object.keys(countBirds).find((key) =>  countBirds[key] === maxBirds)
  return keyBirds
}

console.log(migratoryBirds(birds));